### PR TITLE
Authentication/create new instrument room/navBar,etc

### PR DESCRIPTION
**Pull request by Anoninnyc**

### CODE REVIEW

File: server/db/models.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    type: Sequelize.STRING
+  }
+}, {
+  tableName: 'users',
+  timestamps: false,
+});
+
+sequelize
+  .authenticate()
+  .then(err => {
```
Comment (grrman93) [inline server/db/models.js:74]: did you mean to pass err here as argument?

**Comment by ghost** [comment]
Can you write more descriptive commit messages in the future?

File: client/src/components/LandingPage.jsx [hunk 2]
```diff
@@ -44,8 +44,7 @@ class LandingPage extends React.Component {
     return (
       <div id="landingPage">
 
-        <section className="title">
-          <h1>WELCOME TO TBD!</h1>
+        <section className="title">l
```
Comment (NerdDiffer) [inline client/src/components/LandingPage.jsx:47]: Why is there a `1` here?

File: server/db/models.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    console.log('sqlz Connection has been established successfully.');
+  })
+  .catch(err => {
+    console.log('sqlz Unable to connect to the database:', err);
+  });
+
+sequelize
+  .sync({ force: false })
```
Comment (NerdDiffer) [inline server/db/models.js:82]: We don't want to run `DROP TABLE IF EXISTS` every time. I think we need an init & reset script instead.

File: server/index.js [hunk 4]
```diff
@@ -4,24 +4,85 @@ const path = require('path');
...
 /* Middleware */
-
+app.use(cookieParser());
 app.use(logger('dev'));
-
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
 const pathToStaticDir = path.resolve(__dirname, '..', 'client/public');
 app.use(express.static(pathToStaticDir));
+app.use(expressSession({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
```
Comment (NerdDiffer) [inline server/index.js:27]: the value for `secret` should be an environment variable as well

File: server/index.js [hunk 5]
```diff
@@ -4,24 +4,85 @@ const path = require('path');
...
+  clientID: process.env.client_Id,
+  clientSecret: process.env.client_Secret,
+  callbackURL: "http://localhost:3000/auth/facebook/callback"
+},
+
+  (accessToken, refreshToken, profile, done) => {
+    console.log('this is the profile', profile.id);
+    users.findAll({ where: { facebookId: profile.id }
+  }).then(user => {
+    if (user.map(ind => {
```
Comment (NerdDiffer) [inline server/index.js:41]: Could you please put this condition on 1 line? Better yet, save the value into a variable & use that in the condition.

File: server/index.js [hunk 6]
```diff
@@ -94,12 +155,83 @@ io.on('connection', socket => {
...
+});
+
+app.post('/login', (req, res) => {
+  users.findAll({
+    where: {
+      userName: req.body.user,
+      password: req.body.pass
+    }
+  }).then(user => {
+    if (user.map(ind => {
```
Comment (NerdDiffer) [inline server/index.js:193]: Again, could you please save the condition in a variable & pass that in?

File: server/index.js [hunk 7]
```diff
@@ -94,12 +155,83 @@ io.on('connection', socket => {
...
+});
+
+
+app.post('/signup', (req, res) => {
+  users.findAll({
+    where: {
+      userName: req.body.user
+    }
+  }).then(user => {
+    if (user.map(ind => {
```
Comment (NerdDiffer) [inline server/index.js:214]: Condition in separate variable please, thanks!

File: server/index.js [hunk 8]
```diff
@@ -94,12 +155,83 @@ io.on('connection', socket => {
...
 
 /* Routes */
+app.get('/logout', (req, res) => {
+  console.log('mysession', req.session);
+  if (req.session.userName){
+    delete req.session.userName;
+  }
+  req.logout();
+  console.log('mysession after logout', req.session);
+  res.send('N/A!');
```
Comment (NerdDiffer) [inline server/index.js:183]: Please send back to the client, an appropriate status code. Something like `200` for this case. Here's a good resource for server-side status codes: [HTTP Status Codes](http://www.restapitutorial.com/httpstatuscodes.html)

File: server/index.js [hunk 9]
```diff
@@ -94,12 +155,83 @@ io.on('connection', socket => {
...
+    }
+  });
+});
+
+
+app.get('/auth/facebook', passport.authenticate('facebook'));
+
+app.get('/auth/facebook/callback',
+  passport.authenticate('facebook', { successReturnToOrRedirect: '/', failureRedirect: '/login' }
+      ));
```
Comment (NerdDiffer) [inline server/index.js:237]: please fix indentation

File: server/index.js [hunk 10]
```diff
@@ -4,24 +4,85 @@ const path = require('path');
...
-
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
 const pathToStaticDir = path.resolve(__dirname, '..', 'client/public');
 app.use(express.static(pathToStaticDir));
+app.use(expressSession({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.use(new FacebookStrategy({
```
Comment (NerdDiffer) [inline server/index.js:31]: The indentation is pretty wacked out here. Could you please align starting & ending brackets to the same columns they've started on?

File: client/src/utils/helperFunctions.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+//utils!!!
+import $ from 'jquery';
+
+function showErrorMessage(appendTo,message,id){
+ $(appendTo)
```
Comment (NerdDiffer) [inline client/src/utils/helperFunctions.js:5]: please fix indentation here

File: client/src/utils/helperFunctions.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+        .append('<div id='+ '"'+id+ '"' + '>'+ message+' </div>')
```
Comment (NerdDiffer) [inline client/src/utils/helperFunctions.js:6]: Why not use template strings here?

File: client/src/index.jsx [hunk 13]
```diff
@@ -24,6 +25,7 @@ render((
       <Route path="signup" component={Signup} />
       <Route path="room/:roomId" component={Room} />
       <Route path="metronome" component={Metronome} />
+      <Route path="MakeInstrument" component={UserMakeInstrument} />
```
Comment (NerdDiffer) [inline client/src/index.jsx]: please make the path lower case: `makeinstrument`

File: client/src/components/UserMakeInstrument.jsx [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+              <option value="W">W</option>
+              <option value="E">E</option>
+              <option value="R">R</option>
+              <option value="T">T</option>
+              <option value="Y">Y</option>
+            </select>
+          </form>
+
+        <form>
+        <input id="key_${i}" type="file" name="pic" accept="audio/*" />
```
Comment (NerdDiffer) [inline client/src/components/UserMakeInstrument.jsx]: indent please

File: client/src/components/Signup.jsx [hunk 15]
```diff
@@ -1,16 +1,41 @@
...
 import RaisedButton from 'material-ui/RaisedButton';
+import { Link } from 'react-router';
 import TextField from 'material-ui/TextField';
+import { showErrorMessage } from '../utils/helperFunctions';
 
 class Signup extends Component {
 
+
+  helperSignup() {
+    const user=$('#UNSignUp').val();
```
Comment (NerdDiffer) [inline client/src/components/Signup.jsx:12]: What does this `id` mean? Please choose more descriptive names

File: client/src/components/Login.jsx [hunk 16]
```diff
@@ -1,20 +1,46 @@
...
-        Username:<TextField /><br />
-        Password:<TextField type="password" /><br />
-        <RaisedButton label="Login" onClick={() => { this.props.change('SelectRoom')}} />
-        <RaisedButton label="Click to signup" onClick={() => { this.props.change('SelectRoom')}} />
+      <div id="loginContent">
+        Username:<TextField id="UNLogin" /><br />
+        Password:<TextField id="UNPass" type="password" /><br />
+        <RaisedButton label="Login" onClick={() => { this.helperLogin(); }} / >
+        <Link to="signup"><RaisedButton label="Click to signup" /></Link>
+        <div id="LIMessages"><br /> </div>
```
Comment (NerdDiffer) [inline client/src/components/Login.jsx:30]: Please change to `logInMessages`

File: client/src/components/Login.jsx [hunk 17]
```diff
@@ -1,20 +1,46 @@
...
+
   render() {
     return (
-      <div>
-        Username:<TextField /><br />
-        Password:<TextField type="password" /><br />
-        <RaisedButton label="Login" onClick={() => { this.props.change('SelectRoom')}} />
-        <RaisedButton label="Click to signup" onClick={() => { this.props.change('SelectRoom')}} />
+      <div id="loginContent">
+        Username:<TextField id="UNLogin" /><br />
```
Comment (NerdDiffer) [inline client/src/components/Login.jsx:26]: Please change to something more descriptive. What is `UN` in `UNLogin`?

File: client/src/components/UserMakeInstrument.jsx [hunk 18]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
```
Comment (NerdDiffer) [inline client/src/components/UserMakeInstrument.jsx:1]: For this PR, the `UserMakeInstrument` component is out of scope. Please make a separate PR for this feature

**Comment by NerdDiffer** [comment]
Overall, good work!

File: client/src/components/Login.jsx [hunk 19]
```diff
@@ -1,20 +1,46 @@
...
+        Password:<TextField id="UNPass" type="password" /><br />
```
Comment (ghost) [inline client/src/components/Login.jsx:27]: Can you do `hintText="Hint Text"` and `floatingLabelText="Floating Label Text"` instead of sticking Password: in front

File: client/src/components/Nav/index.jsx [hunk 20]
```diff
@@ -1,26 +1,61 @@
...
+    this.logIn=this.props.logIn.bind(this);
+    this.logOut=this.clearSessions.bind(this);
+    console.log("this.props.user",this.props.user);
+  }
+
+  componentDidMount() {
+      $.get("/fbLoggedIn?", (response, err) => {
+        console.log(response);
+        if (response==="true") {
+          this.logIn("FACEBOOK USER")
```
Comment (ghost) [inline client/src/components/Nav/index.jsx:21]: Why are you just passing in "FACEBOOK USER"?

File: client/src/components/Signup.jsx [hunk 21]
```diff
@@ -1,16 +1,41 @@
...
   render() {
     return (
-      <div>
-        Username:<TextField type="text" /><br />
-        Password:<TextField type="password" /><br />
-        <RaisedButton label="Signup" onClick={() => { this.props.change('selectInstrument')}} />
-        <RaisedButton label="Click to Login Instead" onClick={() => { this.props.change('login')}} />
+      <div id="signupContent">
+
+        Username:<TextField id="UNSignUp" type="text" /><br />
```
Comment (ghost) [inline client/src/components/Signup.jsx:34]: It would be good to tell the user beforehand what the username and password requirements are

File: server/db/models.js (hunk 1 above)
Comment (Anoninnyc) [inline server/db/models.js:74]: done

File: server/index.js (hunk 5 above)
Comment (Anoninnyc) [inline server/index.js:41]: Which condition?

File: server/index.js (hunk 8 above)
Comment (Anoninnyc) [inline server/index.js:183]: done

File: server/index.js (hunk 9 above)
Comment (Anoninnyc) [inline server/index.js:237]: done

File: server/index.js (hunk 10 above)
Comment (Anoninnyc) [inline server/index.js:31]: done

File: client/src/utils/helperFunctions.js (hunk 11 above)
Comment (Anoninnyc) [inline client/src/utils/helperFunctions.js:5]: done

File: client/src/utils/helperFunctions.js (hunk 12 above)
Comment (Anoninnyc) [inline client/src/utils/helperFunctions.js:6]: done-

File: client/src/components/Signup.jsx (hunk 15 above)
Comment (Anoninnyc) [inline client/src/components/Signup.jsx:12]: It means userName..but ill change it..

File: client/src/components/Login.jsx (hunk 17 above)
Comment (Anoninnyc) [inline client/src/components/Login.jsx:26]: done

File: client/src/components/Login.jsx (hunk 16 above)
Comment (Anoninnyc) [inline client/src/components/Login.jsx:30]: changed

File: client/src/components/Login.jsx (hunk 19 above)
Comment (Anoninnyc) [inline client/src/components/Login.jsx:27]: complete

File: client/src/components/Nav/index.jsx (hunk 20 above)
Comment (Anoninnyc) [inline client/src/components/Nav/index.jsx:21]: Placeholder for now. Have yet to pull givenFirstName + givenLastName from FB.

File: client/src/components/Signup.jsx (hunk 21 above)
Comment (Anoninnyc) [inline client/src/components/Signup.jsx:34]: done

**Comment by Anoninnyc** [comment]
got another one....
